// @usedapp/core/dist/test/reducers/notificationsReducer.test.js
"use strict";
exports.__esModule = true;
var chai_1 = require("chai");
var reducer_1 = require("../../src/providers/notifications/reducer");
describe('notificationReducer', function () {
    it('addNotification', function () {
        var notification = {
            id: '1',
            submittedAt: 12,
            transaction: {},
            type: 'transactionStarted'
        };
        chai_1.expect(reducer_1.notificationReducer({}, { chainId: 1, type: 'ADD_NOTIFICATION', notification: notification })).to.deep.equal({
            1: [notification]
        });
    });
    it('notifications added in correct order', function () {
        var initial = {
            id: '1',
            submittedAt: 12,
            transaction: {},
            type: 'transactionStarted'
        };
        var added = {
            id: '2',
            submittedAt: 15,
            transaction: {},
            type: 'transactionStarted'
        };
        var newState = reducer_1.notificationReducer({
            1: [initial]
        }, {
            type: 'ADD_NOTIFICATION',
            chainId: 1,
            notification: added
        });
        chai_1.expect(newState).to.deep.eq({ 1: [added, initial] });
    });
});
